Originally from:
  http://www.math.uni.wroc.pl/~hebisch/gpc/

I believe this patch was posted to gdb-patches and never reviewed.
But M2 support added set types just before GDB 6.5, and now there's
not much left of this.

---
 gdb/dwarf2read.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: gdb-6.6.dfsg/gdb/dwarf2read.c
===================================================================
--- gdb-6.6.dfsg.orig/gdb/dwarf2read.c	2007-01-27 20:35:30.000000000 -0500
+++ gdb-6.6.dfsg/gdb/dwarf2read.c	2007-01-27 20:35:33.000000000 -0500
@@ -6300,9 +6300,11 @@ set_cu_language (unsigned int lang, stru
     case DW_LANG_Modula2:
       cu->language = language_m2;
       break;
+    case DW_LANG_Pascal83:
+      cu->language = language_pascal;
+      break;
     case DW_LANG_Cobol74:
     case DW_LANG_Cobol85:
-    case DW_LANG_Pascal83:
     default:
       cu->language = language_minimal;
       break;
@@ -7496,7 +7498,8 @@ determine_prefix (struct die_info *die, 
   struct die_info *parent;
 
   if (cu->language != language_cplus
-      && cu->language != language_java)
+      && cu->language != language_java
+      && cu->language != language_pascal)
     return NULL;
 
   parent = die->parent;
